services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/VxlanFlowSpec.groovy

```diff
@@ -0,0 +1,396 @@
+            noviflowSwitchesIds.contains(it.dpId)
+        } ?: assumeTrue("Should be at least two active traffgens connected to NoviFlow switches for test execution",
+                false)
+
+        when: "Create a default flow"
+        def defaultFlow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        defaultFlow.source.vlanId = 0
+        defaultFlow.destination.vlanId = 0
+        defaultFlow.encapsulationType = FlowEncapsulationType.VXLAN
+        flowHelper.addFlow(defaultFlow)
+
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.source.vlanId = 10
+        flow.destination.vlanId = 10
+
+        then: "System allows tagged traffic on the default flow"
+        def traffExam = traffExamProvider.get()
+        def exam = new FlowTrafficExamBuilder(topology, traffExam).buildBidirectionalExam(flow, 0)
+        withPool {
+            [exam.forward, exam.reverse].eachParallel { direction ->
+                def resources = traffExam.startExam(direction)
+                direction.setResources(resources)
+                assert traffExam.waitExam(direction).hasTraffic()
+            }
+        }
+
+        and: "Cleanup: Delete the flow"
+        flowHelper.deleteFlow(defaultFlow.id)
+    }
+
+    @Tags(HARDWARE)
+    def "System doesn't allow to create a flow with 'VXLAN' encapsulation on a non-noviflow switches"() {
+        setup:
+        def switchPair = topologyHelper.getAllNeighboringSwitchPairs().find { !it.src.noviflow &&  !it.dst.noviflow }
+        assumeTrue("Unable to find required switches in topology", switchPair as boolean)
+
+        when: "Try to create a flow"
+        def flow = flowHelper.randomFlow(switchPair)
+        flow.encapsulationType = FlowEncapsulationType.VXLAN
+        northbound.addFlow(flow)
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 404
+        // TODO(andriidovhan)fix errorMessage when the 2587 issue is fixed
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Could not create flow: Not enough bandwidth found or path not found. " +
+                "Failed to find path with requested bandwidth=$flow.maximumBandwidth: Switch $switchPair.src.dpId" +
+                " doesn't have links with enough bandwidth"
+    }
+
+    @Tags(HARDWARE)
+    def "System doesn't allow to create a vxlan flow when transit switch is not Noviflow"() {
+        setup:
+        def switchPair = topologyHelper.getAllNotNeighboringSwitchPairs().find {
+            it.src.noviflow && it.dst.noviflow
+        }
+        assumeTrue("Unable to find required switches in topology", switchPair as boolean)
+        def path = switchPair.paths.find { pathHelper.getInvolvedSwitches(it).find { !it.noviflow } }
+        assumeTrue("Unable to find path with required transit switch", path as boolean)
```

Comment:
This is fine for now, but not optimal. Optimal is to include your 'path' requirement to the 'switchPair' search, because it is possible that you pick a novi-novi switch pair that does not have a required path, but there are still other switchPairs out there. Right now you are binding yourself to the first novi-novi pair and if it doesn't have required path - you're done.